pkg/ipam/ipam.go

```diff
@@ -346,6 +348,26 @@ func (i *ipam) allocateIPFromPoolCandidates(ctx context.Context, c *PoolCandidat
 }
 
 func (i *ipam) getPoolCandidates(ctx context.Context, nic string, netConfV4Pool, netConfV6Pool []string, cleanGateway bool, pod *corev1.Pod) ([]*ToBeAllocated, error) {
+	// subnet manager
+	if i.ipamConfig.EnableSubnetManager {
+		_, enableSubnetMgrV4 := pod.Annotations[subnetmanager.AnnoSubnetManagerV4]
+		_, enableSubnetMgrV6 := pod.Annotations[subnetmanager.AnnoSubnetManagerV6]
+		if enableSubnetMgrV4 || enableSubnetMgrV6 {
+			t, err := i.getPoolFromSubnet(ctx, podmanager.GetControllerOwnerType(pod), pod.Namespace, podmanager.GetControllerOwnerName(pod), nic, cleanGateway)
+			if nil != err {
+				return nil, err
+			}
+
+			if t == nil {
+				return nil, fmt.Errorf("pod '%s/%s' specified to use subnet manager but can not found corresponding IPPool", pod.Namespace, pod.Name)
```

Comment:
can not found  --->  can not find

pod '%s/%s' specified to use subnet manager %s  but ...